File: php/src/Behavioral/Template_Method/BookTitleInfo.php
<?

#//BookTitleInfo - two of three concrete templates

require_once 'TitleInfo.php';

class BookTitleInfo extends TitleInfo
{

	private $author;
    
	function __construct($titleName, $author)
	{
		$this->setTitleName($titleName);
		$this->author = $author;
	}

	function setAuthor($a) { $this->author = $a; }
	function getAuthor() { return $this->author; }

	function getTitleBlurb()
	{
		return "Book: " . $this->getTitleName() .  ", Author: " . $this->getAuthor();
	}
}

File: python/src/Behavioral/Template_Method/BookTitleInfo.py

#//BookTitleInfo - two of three concrete templates

from TitleInfo import TitleInfo

class BookTitleInfo(TitleInfo):

	author = None

	def __init__(self,titleName,author) :
		self.setTitleName(titleName)
		self.setAuthor(author)

	def setAuthor (self,a) :
		self.author = a

	def getAuthor (self) : 
		return self.author

	def getTitleBlurb (self): 
		return "Book: {0}, Author: {1}".format( self.getTitleName() , self.getAuthor() )
